src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -4280,18 +4193,24 @@ private String layoutChars(boolean sci) {
             // currency fast path
             int lowInt = (int)intCompact % 100;
             int highInt = (int)intCompact / 100;
-            return (Integer.toString(highInt) + '.' +
-                    StringBuilderHelper.DIGIT_TENS[lowInt] +
-                    StringBuilderHelper.DIGIT_ONES[lowInt]) ;
+            int highIntSize = DecimalDigits.stringSize(highInt);
+            byte[] buf = new byte[highIntSize + 3];
+            DecimalDigits.putPairLatin1(buf, highIntSize + 1, lowInt);
```

Comment:
I think this would read a bit easier if the order of writes to `buf` was reversed